api/src/main/java/org/openmrs/module/fhir2/api/dao/FhirDaoUtils.java

```diff
@@ -0,0 +1,24 @@
+package org.openmrs.module.fhir2.api.dao;
+
+import java.util.Date;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hibernate.criterion.Criterion;
+import org.hibernate.criterion.Restrictions;
+
+public final class FhirDaoUtils {
+	
+	public static Criterion createActiveOrderCriterion(String path) {
+		if (StringUtils.isNotBlank(path)) {
+			path = path + ".";
+		}
+		
+		// ACTIVE = date activated null or less than or equal to current datetime, date stopped null or in the future, auto expire date null or in the future
+		return Restrictions.and(
+		    Restrictions.or(Restrictions.isNull(path + "dateActivated"),
+		        Restrictions.le(path + "dateActivated", new Date())),
+		    Restrictions.or(Restrictions.isNull(path + "dateStopped"), Restrictions.gt(path + "dateStopped", new Date())),
+		    Restrictions.or(Restrictions.isNull(path + "autoExpireDate"),
+		        Restrictions.gt(path + "autoExpireDate", new Date())));
```

Comment:
I'd probably extract out "new Date()), and pass the date into the method.  That would allow you to both a) use the same date for all of the checks, and b) support both "is active now" and "is active at timetime xyz" with the same utility method in the future.

Reply:
Sure,  I can go ahead and do this... I guess my main doubt is that although it's trivial to extract, it's not trivial to set up a bunch of test cases around it, so it will remain untested except for the "today" case for now.